### PR TITLE
Add style guide for contributions

### LINKED ISSUE #62: Add style guide

### PR DESCRIPTION
**Pull request by nvaytet**
Fixes #62

### CODE REVIEW

File: docs/developer/style-guide.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+you should include a ``References`` section at the bottom of your notebook.
+
+Inside this section, you should list the references in alphabatical order,
+and have one markdown cell per entry.
+
+To link from the citation in the notebook's text body to the reference,
+place at the top of each reference cell a ``<div>`` tag with a unique ``id``,
+and make sure to leave a empty line between the ``<div>`` and the reference.
+
+References should be included with the following:
```
Comment (arm61) [inline docs/developer/style-guide.rst:121]: Is it not easier to define a specific reference style?

File: docs/developer/style-guide.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+   [...] as was shown by [Manasi et al. (2021)](#manasi2021).
+
+
+Documentation
+-------------
+
+Apart from workflows in Jupyter notebooks, please provide and update documentation.
+This involves putting python docstrings on your user facing functions,
+providing code comments, and considering  adding other explanations you need to include
```
Comment (arm61) [inline docs/developer/style-guide.rst]: double space

File: docs/developer/style-guide.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+to describe how your functions work.
```
Comment (arm61) [inline docs/developer/style-guide.rst]: mention typing?

**Review by arm61 — APPROVED** [review]

File: docs/developer/style-guide.rst (hunk 3 above)
Comment (nvaytet) [reply docs/developer/style-guide.rst]: You mean the type-hints for the function arguments and return types? yes, good idea

Comment (arm61) [reply docs/developer/style-guide.rst]: that is what I meant sorry

File: docs/developer/style-guide.rst (hunk 1 above)
Comment (nvaytet) [reply docs/developer/style-guide.rst:121]: I didn't really find a know style that did exactly what I wanted... maybe you do?

Comment (arm61) [reply docs/developer/style-guide.rst:121]: Can't we just pick one (APS or RSC or something) and say that the `journal title, issue number and pages` should be a link to the DOI?

File: docs/developer/style-guide.rst [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+Logging
+~~~~~~~
+
+If you have functions in your workflow that represent important steps
+(or costly operations) in your data processing, it is recommended to register the calls
+to those functions via Scipp's logging framework.
+
+Please see the
+`Scipp documentation on logging  <https://scipp.github.io/reference/logging.html>`_
```
Comment (jl-wynen) [inline docs/developer/style-guide.rst]: Instead or in addition, link to https://scipp.github.io/ess/generated/modules/ess.logging.html

File: docs/developer/style-guide.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+If we are aware of failing tests, we can provide future fixes and migrations for you.
+Please avoid large data files, or any code requiring network access.
+Test suites should be fast to execute.
+
+Jupyter notebooks style
+-----------------------
+
+The recommended way to add a data reduction workflow to ``ess`` is to create a Jupyter
+notebook that outlines all the steps in the workflow.
+These notebooks should be added in the ``src/ess/{technique}`` or
```
Comment (jl-wynen) [inline docs/developer/style-guide.rst]: Did you mean the docs folder?

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**